app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/NoteAPI.java

```diff
@@ -119,4 +120,135 @@ public ArrayList<Long> findNotes(String query) {
 
         return noteIds;
     }
-}
+
+    static class NoteInfoField {
+        private final String value;
+        private final int order;
+
+        public NoteInfoField(String value, int order) {
+            this.value = value;
+            this.order = order;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public int getOrder() {
+            return order;
+        }
+    }
+    static class NoteInfo {
+        private final long noteId;
+        private final String modelName;
+        private final List<String> tags;
+        private final Map<String, NoteInfoField> fields;
+
+        public NoteInfo(long noteId, String modelName, List<String> tags, Map<String,
+                NoteInfoField> fields) {
+            this.noteId = noteId;
+            this.modelName = modelName;
+            this.tags = tags;
+            this.fields = fields;
+        }
+
+        public long getNoteId() {
+            return noteId;
+        }
+
+        public String getModelName() {
+            return modelName;
+        }
+
+        public List<String> getTags() {
+            return tags;
+        }
+
+        public Map<String, NoteInfoField> getFields() {
+            return fields;
+        }
+    }
+
+    static class Model {
+        private final long modelId;
+        private final String modelName;
+        private final String[] fieldNames;
+
+        public Model(long modelId, String modelName, String[] fieldNames) {
+            this.modelId = modelId;
+            this.modelName = modelName;
+            this.fieldNames = fieldNames;
+        }
+
+        public long getModelId() {
+            return modelId;
+        }
+
+        public String getModelName() {
+            return modelName;
+        }
+
+        public String[] getFieldNames() {
+            return fieldNames;
+        }
+    }
+
+    public List<NoteInfo> notesInfo(ArrayList<Long> noteIds) throws Exception {
+        List<NoteInfo> notesInfoList = new ArrayList<>();
+        String nidQuery = "nid:" + TextUtils.join(",", noteIds);
+        Map<Long, Model> cache = new HashMap<>();
+
+        Cursor cursor = this.resolver.query(
+                FlashCardsContract.Note.CONTENT_URI,
+                NOTES_INFO_PROJECTION,
+                nidQuery,
+                null,
+                null,
+                null
+                );
+
+        if (cursor == null) {
+            return null;
+        }
+
+        try (cursor) {
+            while (!cursor.moveToNext()) {
```

Comment:
Was this supposed to be `while(cursor.moveToNext())`? It seems like it's ignoring the results of the query right now

Reply:
Thanks for point that out, you are right. 

> Also this is unrelated to your PR but I think it might also just be a minor fix. When an unsupported API is called, default_version() returns a String ("AnkiConnect v.6") and a MalformedJsonException is thrown: https://github.com/killergerbah/asbplayer/issues/408. Since asbplayer also calls addTags, the issue will continue to happen even after notesInfo is added. Is it okay if the response is wrapped like this:

I think for this it would be better if instead of calling default_version() here we just threw an Exception, that way we will respond like the desktop AnkiConnect with a json object of `{"result": null, "error":"unsupported action"}`.

Will test the change in asbplayer to make sure it works before updating PR

Reply:
Sure. Myself and others have been using this branch for some time now since it fixed the duplicate checking bug. Probably only the comment about fixing the result of notesInfo() should be changed before merging.

The other one about changing the result of default_version() should probably be in a different PR and not block this since it's technically unrelated.

Reply:
Sorry, got distracted with work & other things...

I agree with @naiza-asaad on leaving to fix default_version() on a different PR. `notesInfo` logic is fixed in most recent commit.

A few people in TMW have been running the build with these changes for some time now and we haven't heard of any issues, so I think it should be good.

Reply:
This hasn't been resolved by @naiza-asaad for a long while so I will hit resolve, if you have an issue with this or anything else create a new GitHub PR for your changes thanks